login/index.js

```diff
@@ -0,0 +1,71 @@
+import {
+  preventPasswordSpaces,
+  updateButtonState,
+  validateEmailInput,
+  validatePasswordInput,
+  togglePasswordVisiblity,
+} from "../shared/form-utils.js";
+
+const emailInput = document.getElementById("email");
+const emailError = document.getElementById("email-error");
+
+const passwordInput = document.getElementById("password");
+const passwordError = document.getElementById("password-error");
+
+const loginBtn = document.querySelector(".login-btn");
+const loginForm = document.getElementById("LoginForm");
+const eyeIcon = document.getElementById("eyeIcon");
+
+function validateEmail() {
+  return validateEmailInput(emailInput, emailError);
+}
+
+function validatePassword() {
+  return validatePasswordInput(passwordInput, passwordError);
+}
+
+function eyeIconClickHandler() {
+  togglePasswordVisiblity(passwordInput, eyeIcon);
+}
+
+function updateLoginBtnState() {
+  return updateButtonState({
+    inputs: [emailInput, passwordInput],
+    errors: [emailError, passwordError],
+    button: loginBtn,
+  });
+}
+
+emailInput.addEventListener("blur", () => {
+  validateEmail();
+  updateLoginBtnState();
+});
+
+passwordInput.addEventListener("blur", () => {
+  validatePassword();
+  updateLoginBtnState();
+});
+
+emailInput.addEventListener("input", () => {
+  validateEmail();
+  updateLoginBtnState();
+});
+
+passwordInput.addEventListener("input", () => {
+  validatePassword();
+  updateLoginBtnState();
+});
+
+eyeIcon.addEventListener("click", () => {
+  eyeIconClickHandler();
+});
```

Comment:
`eyeIconClickHandler`라는 함수로 한 번 감싸주었기 때문에, 아래처럼 바로 핸들러를 넘겨줘도 됩니다!
```suggestion
eyeIcon.addEventListener('click', eyeIconClickHandler);
```

또는, 래핑 함수를 따로 만들지 않고, 리스너의 콜백에서 처리하는 방법도 있겠네요.
```suggestion
eyeIcon.addEventListener('click', () => {
  togglePasswordVisiblity(passwordInput, eyeIcon);
});
```